headers/validation.h

```diff
@@ -108,6 +109,52 @@ auto operator|(T1 /*unused*/, T2 /*unused*/) {
 
 enum Separator { Space, Newline };
 
+//this contains some specific code which would be available in c++20
+namespace C20 {
+
+constexpr int countl_zero(unsigned long long x) {
+	int res = 64;
+	for (int i = 32; i > 0; i >>= 1) {
+		if ((x >> i) > 0) {
+			res -= i;
+			x >>= i;
+		}
+	}
+	if (x > 0) res--;
+	return res;
+}
+
+int popcount(unsigned long long x) {
+	return static_cast<int>(std::bitset<64>(x).count());
+}
+
+constexpr long double PI = 3.141592653589793238462643383279502884l;
+
+} // namespace C20
+
+namespace Random {
+
+unsigned long long bits64(std::mt19937_64& rng) {
+	static_assert(std::mt19937_64::max() == 0xFFFF'FFFF'FFFF'FFFFull);
+	static_assert(std::mt19937_64::min() == 0ull);
+	return rng();
+}
+
+long double real64(std::mt19937_64& rng) {
```

Comment:
Add a comment that this returns in `[0, 1)` (as opposed to e.g. any positive float)